rdt/transformers/pii/anonymizer.py

```diff
@@ -170,6 +189,13 @@ def _reverse_transform(self, data):
                 f"('{self.get_input_column()}')."
             ) from exception
 
+        if self.missing_value_generation == 'random':
+            nan_mask = np.random.choice(
+                [True, False], size=sample_size,
+                p=[self._nan_frequency, 1 - self._nan_frequency]
+            )
+            reverse_transformed[nan_mask] = np.nan
```

Comment:
we should add a unit test for this part

Reply:
Yes, done in [c8823f1](https://github.com/sdv-dev/RDT/pull/669/commits/c8823f125084af19baaae4a0e66ddac99f6d0acc)